### PR TITLE
created ionic multiselect example

### PR DESCRIPTION
**Pull request by cmreddy071**
Created ionic multiselect-list example
- Written test cases

### CODE REVIEW

File: src/app/app.component.scss [hunk 1]
```diff
@@ -0,0 +1,58 @@
+mat-nav-list{
+  padding-top:0px;
+  mat-checkbox{
+    display: inline-flex;
+  }
+}
+mat-list-icon{
+  vertical-align: center;
+}
+footer{
```
Comment (joebochill) [inline src/app/app.component.scss]: the styles here may need to be updated to be more appropriate for mobile. See if we can find a 'footer' component in ionic that does something similar to SafeAreaView

File: src/app/app.module.ts [hunk 2]
```diff
@@ -1,23 +1,47 @@
...
+import { NgModule, } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { HttpClientModule } from '@angular/common/http';
+import { FlexLayoutModule } from "@angular/flex-layout";
+import { MatButtonModule } from '@angular/material/button';
+import { MatCheckboxModule } from '@angular/material/checkbox';
+import { MatIconModule } from '@angular/material/icon';
+import { MatListModule } from '@angular/material/list';
+import { MatSelectModule } from '@angular/material/select';
```
Comment (joebochill) [inline src/app/app.module.ts]: are we using this?

File: src/app/app.module.ts [hunk 3]
```diff
@@ -1,23 +1,47 @@
...
+import { MatSnackBarModule } from '@angular/material/snack-bar';
+import { MatToolbarModule } from '@angular/material/toolbar';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
```
Comment (joebochill) [inline src/app/app.module.ts]: are we using this?

**Comment by cmreddy071** [comment]
Requested changes are done.
-safe area issue fixed.
-rows are selectable by pressing anywhere on the row
-removed unused imports.

File: src/app/app.component.html [hunk 4]
```diff
@@ -6,35 +6,38 @@
...
+            {{item.name}}
+          </div>
+          <div [ngStyle]="{color: Colors.gray[500]}">
+            {{item.details}}
+          </div>
+        </ion-label>
+        <ion-checkbox slot="start" mode="md" [checked]="isSelected(item.isChecked)" [(ngModel)]="item.isChecked"
+          name="item-{{item.id}}" (ngModelChange)="onSelected(item)"></ion-checkbox>
+      </ion-item>
+    </ion-list>
```
[the remark this answers is not in the document]
Comment (cmreddy071) [reply src/app/app.component.html]: changed to mat-list.

File: src/app/app.component.scss [hunk 5]
```diff
@@ -1,20 +1,10 @@
...
   max-width: 500px;
-  width: 100%;
+  width: 90%;
   position: fixed;
   margin: auto;
-  bottom: 0;
-  left: 0%;  
-  text-align: center;
+  bottom: 0%;
+  left: 0%;    text-align: center;
```
[the remark this answers is not in the document]
Comment (cmreddy071) [reply src/app/app.component.scss]: simplified styles for mobile.

File: src/app/app.component.scss [hunk 6]
```diff
@@ -24,12 +14,15 @@ footer{
   visibility: hidden;
   box-shadow: 0px -8px 28px rgba(0,0,0,0.25), 0px 10px 10px rgba(0,0,0,0.22);
 
+  margin-bottom: constant(safe-area-inset-bottom);
+  margin-bottom: env(safe-area-inset-bottom);
+
```
[the remark this answers is not in the document]
Comment (cmreddy071) [reply src/app/app.component.scss]: removed.

File: src/global.scss [hunk 7]
```diff
@@ -43,4 +43,12 @@ ion-header, ion-toolbar {
 
 .ion-text-center{
   text-align: center
+}
+body {
+  padding-bottom: constant(safe-area-inset-bottom);
+  padding-bottom: env(safe-area-inset-bottom);
```
[the remark this answers is not in the document]
Comment (cmreddy071) [reply src/global.scss]: Removed.

File: src/app/app.component.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (cmreddy071) [reply src/app/app.component.html]: The selection behavior is fixed.

File: src/app/app.component.scss [hunk 8]
```diff
@@ -1,20 +1,10 @@
...
-  }
-}
-mat-list-icon{
-  vertical-align: center;
-}
-footer{
+ion-footer{
   max-width: 500px;
-  width: 100%;
+  width: 90%;
```
[the remark this answers is not in the document]
Comment (cmreddy071) [reply src/app/app.component.scss]: Changed.

File: src/app/app.component.html [hunk 9]
```diff
@@ -1,3 +1,41 @@
...
-</ion-app>
+  <ion-header>
+    <ion-toolbar>
+      <ion-title class="ion-text-center"> Multiselect list </ion-title>
+    </ion-toolbar>
+  </ion-header>
+
+  <ion-content>
+    <mat-nav-list>
+      <mat-list-item *ngFor="let item of data">
```
Comment (joebochill) [inline src/app/app.component.html]: Contents are not properly centered in the rows:
Simulator Screen Shot - iPhone Xʀ - 2019-10-17 at 11 51 01

File: src/app/app.component.scss [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  padding-bottom: 80px;
+  mat-checkbox{
+    display: inline-flex;
+  }
+  .mat-line.title{
+    font-size: 16px;
+  }
+  .mat-line.sub-title{
+    font-size: 14px;
+  }
```
Comment (joebochill) [inline src/app/app.component.scss]: We should not be specifying font styling manually - use Typography

File: src/app/app.component.spec.ts [hunk 11]
```diff
@@ -42,6 +50,20 @@ describe('AppComponent', () => {
...
+    expect(app.enableFooter).toBeFalsy();
+  });
+
+  it('remove all the selected items', async () => {
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.debugElement.componentInstance;
+    app.deleteItems();
+    expect(app.data).toEqual([]);
+    expect(app.enableFooter).toBeFalsy();
+  });
```
Comment (joebochill) [inline src/app/app.component.spec.ts]: Are you calling the deleteItems function with nothing selected?

File: src/app/app.component.spec.ts [hunk 12]
```diff
@@ -42,6 +50,20 @@ describe('AppComponent', () => {
...
   });
 
-  // TODO: add more tests!
+  it('deselect all the selected items', async () => {
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.debugElement.componentInstance;
+    app.cancelItems();
+    expect(app.selectedItems).toEqual([]);
+    expect(app.enableFooter).toBeFalsy();
+  });
```
Comment (joebochill) [inline src/app/app.component.spec.ts]: Are you calling cancelItems with nothing selected?

File: src/global.scss [hunk 13]
```diff
@@ -13,14 +13,66 @@
...
+    color: map-get($pxb-white, 50);
+    --color : map-get($pxb-white, 50);
+    --background: map-get($pxb-blue, 500); 
+    background: map-get($pxb-blue, 500);
+    font-family: Open Sans, sans-serif;    
+}
+
+.ion-text-center{
+  text-align: center
+}
```
Comment (joebochill) [inline src/global.scss]: We probably don't need to define a class for a single css rule if it's only used in one place - just inline it.

File: src/global.scss [hunk 14]
```diff
@@ -13,14 +13,66 @@
...
+    display: inline-block;
+  }
+}
+mat-list-item{
+  .mat-checkbox-inner-container{
+    top: 15px;
+    margin-right: 25px;
+
+  }
+}
```
Comment (joebochill) [inline src/global.scss]: The row should be using flexbox to align it's children, we shouldn't need to use absolute positioning

File: src/global.scss [hunk 15]
```diff
@@ -13,14 +13,66 @@
...
+// body {
+//   padding-bottom: constant(safe-area-inset-bottom);
+//   padding-bottom: env(safe-area-inset-bottom);
+// }
+  
+ion-footer{
+  ion-buttons{
+    width: auto;
+    float: right;
```
Comment (joebochill) [inline src/global.scss]: Why are we floating this? Shouldn't the slot property handle this?

**Comment by cmreddy071** [comment]
Test cases fixed.

File: src/app/app.component.html [hunk 16]
```diff
@@ -1,3 +1,41 @@
...
+        <mat-checkbox mat-list-icon [checked]="isSelected(item)" (change)="onSelected(item)">
+          <div class="title" mat-line [ngStyle]="{color: Colors.black[800]}">
+            {{item.name}}
+          </div>
+          <div class="sub-title" mat-line [ngStyle]="{color: Colors.gray[500]}">
+            {{item.details}}
+          </div>
```
[the remark this answers is not in the document]
Comment (cmreddy071) [reply src/app/app.component.html]: Used angular material typography

File: src/app/app.component.scss (hunk 10 above)
Comment (cmreddy071) [reply src/app/app.component.scss]: used typography for font styling

File: src/app/app.component.spec.ts (hunk 11 above)
Comment (cmreddy071) [reply src/app/app.component.spec.ts]: test cases fixed.

File: src/app/app.component.spec.ts (hunk 12 above)
Comment (cmreddy071) [reply src/app/app.component.spec.ts]: test cases fixed.

File: src/global.scss (hunk 14 above)
Comment (cmreddy071) [reply src/global.scss]: removed.

File: src/global.scss (hunk 13 above)
Comment (cmreddy071) [reply src/global.scss]: Removed.

**Comment by cmreddy071** [comment]
Footer animation fixes.
Float: right is necessary to keep the icons to end.

**Comment by cmreddy071** [comment]
Removed ionic/lab from the dependencies.

**Pull request merged**